sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -435,6 +435,16 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val EXPRESSION_PROJECTION_CANDIDATE_LIMIT =
+    buildConf("spark.sql.optimizer.expressionProjectionCandidateLimit")
+      .doc("The maximum number of the candidate of output expressions whose alias are replaced." +
+        " It can preserve the output partitioning and ordering." +
+        " Negative value means disable this optimization.")
+      .internal()
+      .version("3.4.0")
```

Comment:
This PR targets master, which is `3.5.0`. Is this going to be merged into `branch-3.4`, which is feature-freeze? If not, this line should be adjusted.

Reply:
I'd like to merge it to 3.4 as it fixes a bug in planned write, which is a new feature in 3.4.